bson/src/main/org/bson/ByteBuf.java

```diff
@@ -136,6 +136,26 @@ public interface ByteBuf  {
      */
     byte[] array();
 
+    /**
+     * <p>States whether this buffer is backed by an accessible byte array.</p>
+     *
+     * <p>If this method returns {@code true} then the {@link #array()} and {@link #arrayOffset()} methods may safely be invoked.</p>
+     *
+     * @return {@code true} if, and only if, this buffer is backed by an array and is not read-only
+     * @since 5.5
+     */
+    boolean hasArray();
```

Comment:
optional. I am not able to think of a better method name, but `hasArray()` seems not 100% accurate. Even `backedByArray()` is a little bit better.